### PR TITLE
Add support for lifecycleManagedByParent. 

### LINKED ISSUE #11511: [vscode] Support debug sessions being managed by a parent session
Theia already specifying the parent session in [`DebugSessionOptions`](https://code.visualstudio.com/api/references/vscode-api#DebugSessionOptions) (see also https://github.com/eclipse-theia/theia/pull/9613). However, Theia does not support `DebugSessionOptions.lifecycleManagedByParent`, which controls whether lifecycle requests like 'restart' are sent to the newly created session or its parent session.

Related VS Code API issue:
- #11512

### PR DESCRIPTION
**Pull request by tsmaeder**
#### What it does
Adds support for the `lifecycleSupportedByParent` flag on debug session. Fixes #11511

#### How to test
* Install built-in javascript debugger plugin with version 1.60.1 (https://open-vsx.org/api/ms-vscode/js-debug/1.61.0/file/ms-vscode.js-debug-1.61.0.vsix). 
* Experiment with auto-attached debug sessions: I used Theia itself for testing: it includes both process with many subprocesses and also the launch.json in the repo contains composite launch configurations
* Make sure terminating debug sessions (root sessions as well as descendant sessions) shows the correct behavior as described in https://code.visualstudio.com/docs/editor/debugging#_compound-launch-configurations and https://code.visualstudio.com/api/references/vscode-api#DebugSessionOptions

Contributed on behalf of ST Microelectronics

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/debug/src/browser/debug-session-manager.ts [hunk 1]
```diff
@@ -265,19 +266,25 @@ export class DebugSessionManager {
...
+                compoundRoot?.onDidSessionStop(() => newSession.stop(false, () => this.debug.terminateDebugSession(newSession.id)));
+            }
+            return newSession;
+        }));
         const result = values.every(success => !!success);
         return result;
     }
 
-    protected getCompoundConfigurations(options: DebugCompoundSessionOptions): DebugConfigurationSessionOptions[] {
+    protected getCompoundConfigurations(options: DebugCompoundSessionOptions, compoundRoot: DebugCompoundRoot | undefined): DebugConfigurationSessionOptions[] {
```
[the remark this answers is not in the document]
Comment (tsmaeder) [reply packages/debug/src/browser/debug-session-manager.ts:280]: I want clients to decide whether to pass in a `compoundRoot` or not, therefore the parameter should always be present, even if `undefined`.

**Comment by tsmaeder** [comment]
@msujew could you have another look, please?

**Comment by msujew** [comment]
@tsmaeder I haven't tried reproducing it, but looking at the code, I believe you didn't address the comment I had in [here](https://github.com/eclipse-theia/theia/pull/11751#pullrequestreview-1140701288) yet?

File: packages/debug/src/browser/debug-session-manager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (msujew) [reply packages/debug/src/browser/debug-session-manager.ts:280]: Alright, just wanted to know whether it was a deliberate decision. Fine with me 👍

**Comment by tsmaeder** [comment]
I verified that the code correctly walks upward to the compound configuration an that it terminates all root sessions from there. I was able to reproduce the case where terminating the back end by hand would leave child sessions behind, but not reliably. Looks to me like it's some kind of timing issue that IMO is unrelated to the change at hand.

**Comment by tsmaeder** [comment]
Hmh....turns out the "currentSession" management in the UI is completely off: the session being terminated is not what's selected in the UI. Debugging.

**Comment by tsmaeder** [comment]
Some more debugging shows that the "Launch Browser Frontend" sessions have a child session which has `lifecycleManagedByParent=false`. So going stop going up the chain at this point, which explains why the back end is not completely stopped (just the part that is bound to the front end). Not sure why the session structure in VS Code looks so much different though.

**Comment by tsmaeder** [comment]
@msujew from debugging, I believe our behaviour is correct. Not sure why VS Code behaves differently, but one thing to keep in mind is that it is using a newer version of js-debug than what we are using.

**Comment by tsmaeder** [comment]
I debugged through this case in VS Code: `debugCommands.ts#stopHandler()` has code that walks from the selected debug sessions up the `parentSession` chain to the root: so what is being terminated is the `Launch Browser Frontend` session, not the `editorWorker` sessions. All this leads me to conclude that the code in this PR is correct. Whether we want to change our behaviour to match VS Code as far as which session to terminate when the user presses the "Stop"-button is a question outside the scope of this PR (but I would argue that our behaviour is the better one).

**Review by msujew — APPROVED** [review]
Alright, thank you @tsmaeder for investigating this, I'll now gladly approve this PR :)

I can confirm that the changes work as expected and the property is marked as supported in the API comparison report.

**Pull request merged**